src/screens/Customers/EditCustomerMeter.tsx

```diff
@@ -15,19 +17,24 @@ export interface EditCustomerMeterState {
     meterNumber: string | undefined;
 }
 
-const validationSchema = yup.object({
-    meterType: yup.string().oneOf([MeterType.NO_METER, MeterType.SMART_METER, MeterType.ANALOG_METER], 'Meter Type must be one of Smart, Analog, or No Meter'),
-    meterNumber: yup.mixed() // Can be number or empty string
-        .when('meterType', {
-            is: (MeterType.ANALOG_METER || MeterType.SMART_METER),
-            then: yup.mixed().required('Please enter a meter number'),
-        })
-});
-
 function EditCustomerMeter() {
-
+    const intl = useInternationalization(); 
     const currentCustomer = useSelector(selectCurrentCustomer);
+    const meterTypeMap = new Map<string, string>(); // English keys
+    const revMeterTypeMap = new Map<string, string>(); // Native device language as keys
+    Object.values(MeterType).forEach(meter => {
+        meterTypeMap.set(meter, intl(meter));
+        revMeterTypeMap.set(intl(meter), meter);
+    });
 
+    const validationSchema = yup.object({
+        meterType: yup.string().oneOf([meterTypeMap.get(MeterType.NO_METER), meterTypeMap.get(MeterType.SMART_METER), meterTypeMap.get(MeterType.ANALOG_METER)], intl(words.meter_type_must_be_one_of_smart_analog_or_no_meter)),
+        meterNumber: yup.mixed() // Can be number or empty string
+            .when('meterType', {
+                is: (meterTypeMap.get(MeterType.ANALOG_METER) || meterTypeMap.get(MeterType.SMART_METER)),
+                then: yup.mixed().required(intl(words.please_enter_a_meter_number)),
+            })
+    });
     const formik = useFormik({
         initialValues: {
             meterType: currentCustomer?.meterType,
```

Comment:
`meterType` needs to be mapped here also. So i think it should be ` meterType: currentCustomer?.meterType ? meterTypeMap.get(currentCustomer?.meterType) : ''`